javascripts/discourse/components/reader-mode-options.gjs

```diff
@@ -3,134 +3,147 @@ import { tracked } from "@glimmer/tracking";
+import concatClass from "discourse/helpers/concat-class";
 import icon from "discourse-common/helpers/d-icon";
 import DMenu from "float-kit/components/d-menu";
 
 export default class ReaderModeOptions extends Component {
+  @service readerMode;
+
   @tracked showOptions = false;
 
   @action
   toggleOptions() {
     this.showOptions = !this.showOptions;
   }
 
-  @action
-  selectFont(e) {
-    document.documentElement.style.setProperty(
-      "--reader-mode-font-family",
-      e.target.value
-    );
-  }
-
-  @action
-  selectFontSize(e) {
-    const setFontSize = (property, baseValue) => {
-      const value =
-        e.target.value === 0 ? baseValue : baseValue + e.target.value * 0.1;
-      document.documentElement.style.setProperty(property, `${value}rem`);
-    };
-
-    setFontSize("--reader-mode-font-size", 1);
-    setFontSize("--reader-mode-h1-font-size", 1.517);
-    setFontSize("--reader-mode-h2-font-size", 1.3195);
-    setFontSize("--reader-mode-h3-font-size", 1.1487);
-    setFontSize("--reader-mode-h4-font-size", 1);
-    setFontSize("--reader-mode-h5-font-size", 0.8706);
-    setFontSize("--reader-mode-h6-font-size", 0.7579);
-    setFontSize("--reader-mode-small-font-size", 0.75);
-    setFontSize("--reader-mode-big-font-size", 1.5);
-  }
-
-  @action
-  selectOffset(e) {
-    let contentWidthVariable = `-${e.target.value}px`;
-
-    let topicBodyWidth = parseInt(
-      window
-        .getComputedStyle(document.documentElement)
-        .getPropertyValue("--topic-body-width"),
-      10
-    );
-    let topicbodyWidthPadding = parseInt(
-      window
-        .getComputedStyle(document.documentElement)
-        .getPropertyValue("--topic-body-width-padding"),
-      10
-    );
-    let additionalWidth = parseInt(e.target.value, 10);
-    let topicBodyWidthVariable = `${
-      topicBodyWidth + additionalWidth + topicbodyWidthPadding * 2
-    }px`;
-
-    // Move the content to the left when width increases
-    document.documentElement.style.setProperty(
-      "--reader-mode-offset",
-      contentWidthVariable
-    );
-    // increase with of topic body
-    document.documentElement.style.setProperty(
-      "--reader-mode-topic-body-width",
-      topicBodyWidthVariable
-    );
-    // increase defined grid width for topic content
-    document.documentElement.style.setProperty(
-      "--reader-mode-topic-grid",
-      `${this.args.topicGridWidth + parseInt(e.target.value, 10)}px ${
-        this.args.timelineGridWidth
-      }px`
-    );
-  }
-
   <template>
     <DMenu
       @identifier="reader-mode-options"
       @triggers={{array "click"}}
       @placementStrategy="fixed"
       @class="reader-mode-options"
       @inline={{true}}
+      {{didInsert this.readerMode.selectFontSize}}
+      {{didInsert this.readerMode.selectFont}}
+      {{didInsert this.readerMode.selectOffset}}
+      {{didInsert this.readerMode.setupWidth}}
+      {{didInsert this.readerMode.setupColors}}
```

Comment:
Not necessary, but we could move these to a single function for a single `didInsert`.

Eg.

```suggestion
      {{didInsert this.setupReaderMode}}
```
```js
setupReaderMode(){
   this.readerMode.selectFontSize()
   ..etc
}
```

Reply:
Love that idea